homeassistant/components/sensor/__init__.py

```diff
@@ -505,13 +507,33 @@ def native_unit_of_measurement(self) -> str | None:
             return self.entity_description.native_unit_of_measurement
         return None
 
+    @property
+    def display_unit_of_measurement(self) -> str | None:
```

Comment:
Display is not a good word as it's not only for display. We use this unit in the state machine. The state can be consumed in many different ways.